### PR TITLE
[FLINK-15567][table][docs] Add documentation for INSERT statements for Flink SQL

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

We missed to add documentation for INSERT statements which should be added under "SQL" page.

## Brief change log

Add `insert.md` and `insert.zh.md` under `sql` directory.

## Verifying this change

No tests required. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @JingsongLi

File: docs/dev/table/sql/insert.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+CREATE TABLE country_page_view (user STRING, cnt INT, date STRING, country STRING)
+PARTITIONED BY (date, country)
+WITH (...)
+
+-- Appends rows into the partition (date='2019-8-30', country='China')
+INSERT INTO country_page_view PARTITION (date='2019-8-30', country='China')
+  SELECT user, cnt FROM page_view_source;
+
+-- Overwrites the partition (date='2019-8-30', country='China') using rows in page_view_source
+INSERT INTO country_page_view PARTITION (date='2019-8-30', country='China')
```
Comment (JingsongLi) [inline docs/dev/table/sql/insert.md]: Can you add append/overwrite dynamic partition inserting?

File: docs/dev/table/sql/insert.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+INSERT statements are used to add rows to a table.
+
+## Run an INSERT statement
+
+INSERT statements are specified with the `sqlUpdate()` method of the `TableEnvironment` or executed in [SQL CLI]({{ site.baseurl }}/dev/table/sqlClient.html). The method `sqlUpdate()` for INSERT statements is a lazy execution, they will be executed only when `TableEnvironment.execute(jobName)` is invoked.
```
[the remark this answers is not in the document]
Comment (wuchong) [reply docs/dev/table/sql/insert.md]: Why should we emphasize `detach` mode here? It is not related to the `sqlUpdate()` problem.

File: docs/dev/table/sql/insert.md (hunk 1 above)
Comment (wuchong) [reply docs/dev/table/sql/insert.md]: Sure.

File: docs/dev/table/sql/insert.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JingsongLi) [reply docs/dev/table/sql/insert.md]: ok

File: docs/dev/table/sql/insert.md [hunk 3]
```diff
@@ -125,20 +125,28 @@ part_spec:
...
 
--- Appends rows into the partition (date='2019-8-30', country='China')
+-- Appends rows into the static partition (date='2019-8-30', country='China')
 INSERT INTO country_page_view PARTITION (date='2019-8-30', country='China')
   SELECT user, cnt FROM page_view_source;
 
--- Overwrites the partition (date='2019-8-30', country='China') using rows in page_view_source
+-- Appends rows into partition (date, country), where date is static partition with value '2019-8-30',
+-- country is dynamic partition whose value is dynamic determined by each row.
 INSERT INTO country_page_view PARTITION (date='2019-8-30', country='China')
```
Comment (JingsongLi) [inline docs/dev/table/sql/insert.md]: Remove `, country='China'`

**Comment by JingsongLi** [comment]
LGTM

**Comment by wuchong** [comment]
Thanks @JingsongLi , Merging...

**Pull request merged**